### PR TITLE
auto redeploy the service if the container is unhealthy and auto redeploy is enabled

### PR DESCRIPTION
**Pull request by liyaqin1**
Option for auto restart in serviceset config/ healthcheck
In deploy agent, restart the required times when deployment fails.
Related PRs: 
https://github.com/pinternal/teletraan-config-manager/pull/95 
https://github.com/pinternal/teletraan-config-manager/pull/101

### CODE REVIEW

**Comment by liyaqin1** [comment]
Tested together with these two PRs https://github.com/pinternal/teletraan-config-manager/pull/95 and 
https://github.com/pinternal/teletraan-config-manager/pull/101 on my devapp. Looks good.
The Telefig conf file I used:
```
version: 1
tcollector:
  docker:
    version: abe3077cb493409e5265de35888389578d39ae9e
    healthcheck:
      http: http://localhost:18132
      interval: 10
      timeout: 1
      start_period: 5
      retries: 3
      redeploy_when_unhealthy: True
      redeploy_retry_times: 3
```

Test case 1: 
I manually trigggered a unhealthy tcollector and then ran this PR as teletraan deployd code, I see the file /mnt/deployd/tcollector was generated and the content in it is 1. And the tcollector item was deleted and was redeploy again using the correct conf. So, it is back to healthy again. Then I ran this PR as deployd code, the content in file /mnt/deployd/tcollector is 0 which is expected.

Test case 2:
I manually trigggered a unhealthy tcollector and then ran this PR as teletraan deployd code when the retry times reached 3 in the file, it didn't redeploy anymore. 

Test case 3:
When the tcollector is in healthy status, running this PR didn't cause any redeploy.

File: deploy-agent/deployd/common/utils.py [hunk 1]
```diff
@@ -236,7 +236,25 @@ def get_container_health_info(commit):
                         command = ['docker', 'inspect', '-f', '{{.State.Health.Status}}', name]
                         status = subprocess.run(command, check=True, stdout=subprocess.PIPE).stdout
                         if status:
-                            result.append(f"{name}:{status.decode().strip()}")
+                            status = status.decode().strip()
+                            if status == "unhealthy" and "redeploy_when_unhealthy=enabled" in parts[2]:
+                                labels = parts[2].split(',')
+                                max_retry = REDEPLOY_RETRY_TIMES
+                                for label in labels:
+                                    if "redeploy_retry_times" in label:
```
Comment (ntascii) [inline deploy-agent/deployd/common/utils.py:244]: let's get the retry into a helper function

File: deploy-agent/deployd/common/utils.py [hunk 2]
```diff
@@ -236,7 +236,25 @@ def get_container_health_info(commit):
...
+                                        max_retry = int(label.split('=')[1])
+                                retry_num = 0
+                                file_name = "/mnt/deployd/" + service
+                                if os.path.exists(file_name):
+                                    f=open(file_name,"r")
+                                    retry_num = int(f.readline())
+                                    f.close()
+                                if retry_num < max_retry:
+                                    ff=open(file_name,"w")
```
Comment (ntascii) [inline deploy-agent/deployd/common/utils.py:253]: `with open(file_name, mode="w") as file:`

Comment (liyaqin1) [reply deploy-agent/deployd/common/utils.py:253]: sure, thank you

File: deploy-agent/deployd/common/utils.py (hunk 1 above)
Comment (liyaqin1) [reply deploy-agent/deployd/common/utils.py:244]: sure, thank you

**Review by ntascii — APPROVED** [review]

**Pull request merged**